subsys/bluetooth/host/hci_core.c

```diff
@@ -7123,6 +7300,335 @@ bool bt_addr_le_is_bonded(uint8_t id, const bt_addr_le_t *addr)
+	if (!atomic_test_bit(adv->flags, BT_PER_ADV_PARAMS_SET)) {
+		return -EINVAL;
+	}
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_SET_PER_ADV_ENABLE, sizeof(*cp));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	(void)memset(cp, 0, sizeof(*cp));
+
+	cp->handle = adv->handle;
+	cp->enable = enable ? 1 : 0;
+
+
+	cmd_state_set_init(&state, adv->flags, BT_PER_ADV_ENABLED, enable);
+	cmd(buf)->state = &state;
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_LE_SET_PER_ADV_ENABLE, buf, NULL);
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+int bt_le_per_adv_start(struct bt_le_ext_adv *adv)
+{
+	return bt_le_per_adv_enable(adv, true);
+}
+
+int bt_le_per_adv_stop(struct bt_le_ext_adv *adv)
+{
+	return bt_le_per_adv_enable(adv, false);
+}
+
+#if defined(CONFIG_BT_PER_ADV_SYNC)
+
+uint8_t bt_le_per_adv_sync_get_index(struct bt_le_per_adv_sync *per_adv_sync)
+{
+	uintptr_t index = per_adv_sync - per_adv_sync_pool;
+
+	__ASSERT(per_adv_sync >= per_adv_sync_pool &&
+			index < ARRAY_SIZE(per_adv_sync_pool),
+		 "Invalid per_adv_sync pointer");
+	return index;
+}
+
+static struct bt_le_per_adv_sync *per_adv_sync_new(void)
+{
+	struct bt_le_per_adv_sync *per_adv_sync = NULL;
+
+	for (int i = 0; i < ARRAY_SIZE(per_adv_sync_pool); i++) {
+		if (!atomic_test_bit(per_adv_sync_pool[i].flags,
+				     BT_PER_ADV_SYNC_CREATED)) {
+			per_adv_sync = &per_adv_sync_pool[i];
+			break;
+		}
+	}
+
+	if (!per_adv_sync) {
+		return NULL;
+	}
+
+	(void)memset(per_adv_sync, 0, sizeof(*per_adv_sync));
+	atomic_set_bit(per_adv_sync->flags, BT_PER_ADV_SYNC_CREATED);
+
+	return per_adv_sync;
+}
+
+static void per_adv_sync_delete(struct bt_le_per_adv_sync *per_adv_sync)
+{
+	atomic_clear(per_adv_sync->flags);
+	per_adv_sync->cb = NULL; /* disable callbacks */
+}
+
+int bt_le_per_adv_sync_create(const struct bt_le_per_adv_sync_param *param,
+			      const struct bt_le_per_adv_sync_cb *cb,
+			      struct bt_le_per_adv_sync **out_sync)
+{
+	struct bt_hci_cp_le_per_adv_create_sync *cp;
+	struct net_buf *buf;
+	struct bt_le_per_adv_sync *per_adv_sync;
+	int err;
+
+	if (!BT_FEAT_LE_EXT_PER_ADV(bt_dev.le.features)) {
+		return -ENOTSUP;
+	}
+
+	if (get_pending_per_adv_sync()) {
+		return -EBUSY;
+	}
+
+	if (param->sid > BT_GAP_SID_MAX ||
+		   param->skip > BT_GAP_PER_ADV_MAX_MAX_SKIP ||
+		   param->timeout > BT_GAP_PER_ADV_MAX_MAX_TIMEOUT) {
+		return -EINVAL;
+	}
+
+	if (atomic_test_bit(bt_dev.flags, BT_DEV_EXPLICIT_SCAN)) {
+		return -EBUSY;
+	}
+
+	per_adv_sync = per_adv_sync_new();
+	if (!per_adv_sync) {
+		return -ENOMEM;
+	}
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_PER_ADV_CREATE_SYNC, sizeof(*cp));
+	if (!buf) {
+		per_adv_sync_delete(per_adv_sync);
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	(void)memset(cp, 0, sizeof(*cp));
+
+
+	bt_addr_le_copy(&cp->addr, &param->addr);
+
+	if (param->options & BT_LE_PER_ADV_SYNC_OPT_USE_PER_ADV_LIST) {
+		cp->filter_policy |= BT_HCI_LE_PER_ADV_CREATE_SYNC_FP_USE_LIST;
```

Comment:
This parameter is called `options` in the 5.2 spec, I think we should rename it to match.

Reply:
Agreed